neurom/core/morphology.py

```diff
@@ -288,7 +292,10 @@ def iter_sections(neurites,
     )
 
 
-def iter_segments(obj, neurite_filter=None, neurite_order=NeuriteIter.FileOrder):
+def iter_segments(obj,
```

Comment:
Could you please add a section_filter now that you are at it?

See the implementation on the v4 branch: 
https://github.com/BlueBrain/NeuroM/blob/41ec17f3cb93166a19294117927ee1d10e75075e/neurom/core/morphology.py#L354

Or you could add the changes directly to v4 if they are not urgent.